lib/internal/modules/esm/get_format.js

```diff
@@ -82,6 +84,24 @@ function underNodeModules(url) {
   return StringPrototypeIncludes(url.pathname, '/node_modules/');
 }
 
+/**
+ * Determine whether the given source contains CJS or ESM module syntax.
+ * @param {string} source
+ * @param {URL} url
+ */
+function detectModuleFormat(source, url) {
+  try {
+    let realSource = source ?? readFileSync(url, 'utf8');
+    if (Buffer.isBuffer(realSource)) {
+      // `containsModuleSyntax` requires source to be passed in as string
+      realSource = realSource.toString();
+    }
```

Comment:
If I remove the following lines, all tests still pass. I suggest we remove them (unless we can have a test, but maybe it should be its own PR).
```suggestion
```